{"ast":null,"code":"\"use strict\";\n\nvar __extends = this && this.__extends || function () {\n  var extendStatics = function (d, b) {\n    extendStatics = Object.setPrototypeOf || {\n      __proto__: []\n    } instanceof Array && function (d, b) {\n      d.__proto__ = b;\n    } || function (d, b) {\n      for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];\n    };\n\n    return extendStatics(d, b);\n  };\n\n  return function (d, b) {\n    extendStatics(d, b);\n\n    function __() {\n      this.constructor = d;\n    }\n\n    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());\n  };\n}();\n\nObject.defineProperty(exports, \"__esModule\", {\n  value: true\n});\n\nvar ts_custom_error_1 = require(\"ts-custom-error\");\n/**\n * Custom Error class of type Exception.\n */\n\n\nvar Exception =\n/** @class */\nfunction (_super) {\n  __extends(Exception, _super);\n  /**\n   * Allows Exception to be constructed directly\n   * with some message and prototype definition.\n   */\n\n\n  function Exception(message) {\n    if (message === void 0) {\n      message = undefined;\n    }\n\n    var _this = _super.call(this, message) || this;\n\n    _this.message = message;\n    return _this;\n  }\n\n  return Exception;\n}(ts_custom_error_1.CustomError);\n\nexports.default = Exception;","map":{"version":3,"sources":["../../src/core/Exception.ts"],"names":[],"mappings":";;;;;;;;;;;;;;;;;;;;;;;;;;;;;;AAAA,IAAA,iBAAA,GAAA,OAAA,CAAA,iBAAA,CAAA;AAEA;;AAEG;;;AACH,IAAA,SAAA;AAAA;AAAA,UAAA,MAAA,EAAA;AAAuC,EAAA,SAAA,CAAA,SAAA,EAAA,MAAA,CAAA;AAErC;;;AAGG;;;AACH,WAAA,SAAA,CACS,OADT,EACoC;AAA3B,QAAA,OAAA,KAAA,KAAA,CAAA,EAAA;AAAA,MAAA,OAAA,GAAA,SAAA;AAA2B;;AADpC,QAAA,KAAA,GAGE,MAAA,CAAA,IAAA,CAAA,IAAA,EAAM,OAAN,KAAc,IAHhB;;AACS,IAAA,KAAA,CAAA,OAAA,GAAA,OAAA;;AAGR;;AACH,SAAA,SAAA;AAAC,CAXD,CAAuC,iBAAA,CAAA,WAAvC,CAAA","sourceRoot":"","sourcesContent":["\"use strict\";\nvar __extends = (this && this.__extends) || (function () {\n    var extendStatics = function (d, b) {\n        extendStatics = Object.setPrototypeOf ||\n            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||\n            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };\n        return extendStatics(d, b);\n    };\n    return function (d, b) {\n        extendStatics(d, b);\n        function __() { this.constructor = d; }\n        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());\n    };\n})();\nObject.defineProperty(exports, \"__esModule\", { value: true });\nvar ts_custom_error_1 = require(\"ts-custom-error\");\n/**\n * Custom Error class of type Exception.\n */\nvar Exception = /** @class */ (function (_super) {\n    __extends(Exception, _super);\n    /**\n     * Allows Exception to be constructed directly\n     * with some message and prototype definition.\n     */\n    function Exception(message) {\n        if (message === void 0) { message = undefined; }\n        var _this = _super.call(this, message) || this;\n        _this.message = message;\n        return _this;\n    }\n    return Exception;\n}(ts_custom_error_1.CustomError));\nexports.default = Exception;\n//# sourceMappingURL=Exception.js.map"]},"metadata":{},"sourceType":"script"}